src/crypto/algorithms/megolm.js

```diff
@@ -262,15 +270,162 @@ MegolmEncryption.prototype._prepareNewSession = async function() {
  *
  * @param {module:crypto/algorithms/megolm.OutboundSessionInfo} session
  *
+ * @param {object<userId, deviceId>} devicemap
+ *   mapping from userId to deviceId to {@link module:crypto~OlmSessionResult}
+ *
  * @param {object<string, module:crypto/deviceinfo[]>} devicesByUser
  *    map from userid to list of devices
  *
+ * @return {array<object<userid, deviceInfo>>}
+ */
+MegolmEncryption.prototype._splitUserDeviceMap = function(
+    session, devicemap, devicesByUser,
+) {
+    const maxToDeviceMessagesPerRequest = 20;
+
+    // use an array where the slices of a content map gets stored
+    const contentMaps = [];
+    let maxContentMapId = 0; // start inserting in the first contentMap
+    let currentToDeviceId = 0;
+
+    const key = this._olmDevice.getOutboundGroupSessionKey(session.sessionId);
```

Comment:
suggest you pass `chain_index` as a param alongside `session` rather than calling this again